src/algebra/group_power/lemmas.lean

```diff
@@ -205,6 +205,14 @@ calc n • a = n • a + 0 : (add_zero _).symm
   ... < n • a + (m - n) • a : add_lt_add_left (gsmul_pos ha (sub_pos.mpr h)) _
   ... = m • a : by { rw [← add_gsmul], simp }
 
+lemma gsmul_lt_gsmul_of_lt_right_of_pos {a b : A} {m : ℤ} (hab : a < b) (hm : 0 < m) :
```

Comment:
Should this be `gsmul_strict_mono` and use the `strict_mono` machinery?
There could then also be a version for `m < 0`. The `strict_mono` API will make the injectivity lemma below a 1-liner.

Reply:
That's a good idea!

Reply:
I couldn't get this down to a one-linter, but I was able to eliminate everything after the `suffices`